cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceCheckerTest.java

```diff
@@ -0,0 +1,147 @@
+
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import javax.ws.rs.core.Response;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseException;
+import com.sequenceiq.cloudbreak.cloud.gcp.GcpResourceException;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@ExtendWith(MockitoExtension.class)
+class AbstractGcpBaseResourceCheckerTest {
+
+    private static final String RESOURCE_NAME = "something";
+
+    @Mock
+    private GoogleJsonResponseException mockGoogleJsonResponseException;
+
+    private AbstractGcpBaseResourceChecker underTest;
+
+    @BeforeEach
+    void setUp() {
+        underTest = new AbstractGcpBaseResourceChecker() {
+            @Override
+            protected void exceptionHandler(GoogleJsonResponseException ex, String name, ResourceType resourceType) {
+                super.exceptionHandler(ex, name, resourceType);
+            }
+        };
+    }
+
+    @ParameterizedTest
+    @EnumSource(ResourceType.class)
+    void testExceptionHandlerWhenGoogleJsonResponseExceptionIsNullThenIllegalArgumentExceptionShouldCome(ResourceType resourceType) {
+        assertThrows(IllegalArgumentException.class, () -> underTest.exceptionHandler(null, RESOURCE_NAME, resourceType));
+    }
+
+    @ParameterizedTest
+    @EnumSource(ResourceType.class)
+    void testExceptionHandlerWhenGoogleJsonErrorIsNullThenGcpResourceExceptionShouldCome(ResourceType resourceType) {
+        String expectedExceptionMessage = "somethingAwful";
+        when(mockGoogleJsonResponseException.getMessage()).thenReturn(expectedExceptionMessage);
+        GcpResourceException expectedException = assertThrows(GcpResourceException.class,
+                () -> underTest.exceptionHandler(mockGoogleJsonResponseException, RESOURCE_NAME, resourceType));
+
+        assertNotNull(expectedException.getMessage());
+        assertTrue(expectedException.getMessage().contains(expectedExceptionMessage));
```

Comment:
consider using AssertJ, it provides more usable assert fail messages (and has substring checks as well)

Reply:
used AssertJ